encoding/kzgrs/prover/prover.go

```diff
@@ -0,0 +1,280 @@
+package prover
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"math"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/Layr-Labs/eigenda/encoding/kzgrs"
+	"github.com/Layr-Labs/eigenda/encoding/rs"
+	kzg "github.com/Layr-Labs/eigenda/pkg/kzg"
+
+	bls "github.com/Layr-Labs/eigenda/pkg/kzg/bn254"
+	_ "go.uber.org/automaxprocs"
+)
+
+type Prover struct {
+	*kzgrs.KzgConfig
+	Srs          *kzg.SRS
+	G2Trailing   []bls.G2Point
+	mu           sync.Mutex
+	LoadG2Points bool
+
+	ParametrizedProvers map[rs.EncodingParams]*ParametrizedProver
+}
+
+func NewProver(config *kzgrs.KzgConfig, loadG2Points bool) (*Prover, error) {
```

Comment:
Why not name this as ProverFactory as previously discussed? I think it fits quite better.

Reply:
I think the next PR will make this clearer. We may make the low level factory methods private and only expose the high level interface methods.